src/components/number-picker/number-picker.component.ts

```diff
@@ -85,11 +86,15 @@ export class NumberPickerComponent implements ControlValueAccessor, OnDestroy, O
     /** Defines the amount the number picker should increase or decrease when the buttons or arrow keys are used. */
     @Input()
     get step(): number {
-        return this._step;
+        return typeof this._step === 'number' ? this._step : this._step(this.value, this._direction);
```

Comment:
I don't think this is quite right, the getter should be the same type as the setter. I'd suggest getting rid of the getter and set up a function with a `direction` arg to compute the step value.